amazon-kinesis-client/src/test/java/software/amazon/kinesis/lifecycle/ShardConsumerSubscriberTest.java

```diff
@@ -311,6 +309,80 @@ public void restartAfterRequestTimerExpiresTest() throws Exception {
 
     }
 
+    @Test
+    public void restartAfterRequestTimerExpiresAfterInitialSubscriptionFailsTest() throws Exception {
+
+        executorService = Executors.newFixedThreadPool(1, new ThreadFactoryBuilder()
+                .setNameFormat("test-" + testName.getMethodName() + "-%04d").setDaemon(true).build());
+
+        recordsPublisher = new RecordPublisherWithInitialFailureSubscription();
+        subscriber = new ShardConsumerSubscriber(recordsPublisher, executorService, bufferSize, shardConsumer, 0);
+        addUniqueItem(1);
+        addTerminalMarker(1);
+
+        CyclicBarrier barrier = new CyclicBarrier(2);
+
+        List<ProcessRecordsInput> received = new ArrayList<>();
+        doAnswer(a -> {
+            ProcessRecordsInput input = a.getArgumentAt(0, ProcessRecordsInput.class);
+            received.add(input);
+            if (input.records().stream().anyMatch(r -> StringUtils.startsWith(r.partitionKey(), TERMINAL_MARKER))) {
+                synchronized (processedNotifier) {
+                    processedNotifier.notifyAll();
+                }
+            }
+            return null;
+        }).when(shardConsumer).handleInput(any(ProcessRecordsInput.class), any(Subscription.class));
+
+        synchronized (processedNotifier) {
+            subscriber.startSubscriptions();
+            processedNotifier.wait(1000);
+        }
+
+        Stream.iterate(2, i -> i + 1).limit(97).forEach(this::addUniqueItem);
+
+        addTerminalMarker(2);
+
+        verify(shardConsumer, times(0)).handleInput(argThat(eqProcessRecordsInput(processRecordsInput)),
```

Comment:
If you're trying to verify that the shardConsumer doesn't initialize successfully, might it be clearer to use the `never()` matcher rather than `timer(0)`?

Reply:
i didn't knew about never() , new to mockito always been a EasyMock user :)